lib/shared/src/token/counter.ts

```diff
@@ -67,11 +67,14 @@ export async function getTokenCounterUtils(): Promise<TokenCounterUtils> {
                     },
 
                     countTokens(text: string): number {
-                        return tokenCounterUtils.encode(text).length
+                        const wordCount = text.split(' ').length
```

Comment:
```suggestion
                        const wordCount = text.trim().split(/\s+/).filter(Boolean);
```
To consider:

- Multiple consecutive spaces
- Tabs and newlines
- Leading/trailing whitespace
- Empty strings
- Unicode whitespace characters (including ideographic spaces)